### PR TITLE
WIP: Ui razor 2 class -

### LINKED ISSUE #5409: UI as a library - Package Razor pages and views as reusable libraries
aspnet/Razor#1809

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #5409 
closed and moved to #5980

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@rynowak are you the right person to review Razor UI in class library?

**Comment by Rick-Anderson** [comment]
@rynowak double check the adding reference section. Did I list the right DLLs? BTW, I couldn't get this working with DLLs or my NuGet package.

**Comment by rynowak** [comment]
@javiercn and @pranavkm are the right people to review this

File: aspnetcore/mvc/razor-pages/ui-class.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: advanced
+uid: mvc/razor-pages/ui-class
+---
+# Reusable Razor User Interface (UI) in class libraries with ASP.NET Core
+
+By [Rick Anderson](https://twitter.com/RickAndMSFT)
+
+Razor views, pages, controllers, page models, and data models can be packaged and shared in a class library. Applications can include the Razor UI class library and override the views and pages it contains.
```
Comment (pranavkm) [inline aspnetcore/mvc/razor-pages/ui-class.md]: can be built in a class library project, and packaged and re-used (?).

File: aspnetcore/mvc/razor-pages/ui-class.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+* Create a .NET Standard class library. For example, run 
+
+    ```cli
+    dotnet new classlib -f netstandard2.0 -o RazorClassUI
+    ```
+    See [dotnet new](/dotnet/core/tools/dotnet-new) for more information.
+* Update the *.csproj* file to contain:
+
+    * The MSBuild properties `ResolvedRazorCompileToolset`, `RazorCompileOnBuild`, `IncludeContentInPack`.
```
Comment (pranavkm) [inline aspnetcore/mvc/razor-pages/ui-class.md]: Ideally we want people to use the preview2 sdk where 

a) None of these properties are required
b) You get tooling for Razor.

Should we draft this for the preview2 Sdk and release the docs once it's available? The instructions are a bit different and lot more streamlined.

**Pull request closed without merging**